consensus/parlia/parlia.go

```diff
@@ -1538,11 +1591,26 @@ func (p *Parlia) getCurrentValidators(blockHash common.Hash, blockNum *big.Int)
 		return nil, nil, err
 	}
 
-	voteAddrmap := make(map[common.Address]*types.BLSPublicKey, len(valSet))
+	voteAddrMap := make(map[common.Address]*types.BLSPublicKey, len(valSet))
+	voteAddrExist := make(map[types.BLSPublicKey]bool, len(valSet))
+
+	// after FusionBlock, there may be same validator from BC and BSC
+	// we should remove duplicated consensus address and vote address
+	// if duplicated, use the first one which has more voting power
 	for i := 0; i < len(valSet); i++ {
-		voteAddrmap[valSet[i]] = &(voteAddrSet)[i]
+		key := valSet[i]
+		value := voteAddrSet[i]
+
+		if _, ok := voteAddrMap[key]; !ok {
```

Comment:
because `voteAddrMap` is empty at the begining, so it will be always "!ok" and continue, and thus never go to `voteAddrMap[key] = &value`?

Reply:
Done